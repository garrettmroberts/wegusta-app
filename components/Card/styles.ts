import { StyleSheet } from 'react-native'
import Colors from '../../constants/Colors'
import Sizes from '../../constants/Sizes'

const styles = StyleSheet.create({
  cardContainer: {
    backgroundColor: Colors.white,
    height: Sizes.cardHeight,
    borderRadius: 20,
    overflow: 'hidden'
  },
  card: {
    height: Sizes.cardHeight,
    width: Sizes.screenWidth - Sizes.viewPadding * 2,
    justifyContent: 'center'
  },
  image: {
    flex: 1,
    resizeMode: 'cover',
    overflow: 'hidden'
  }
})

export default styles
