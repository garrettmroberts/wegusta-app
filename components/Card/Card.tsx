import { ImageBackground, ImageSourcePropType, View } from 'react-native';

import styles from './styles';

type Props = {
  imageProps: ImageSourcePropType
}

const Card = ({ imageProps }: Props) => {
  return (
    <View style={styles.card}>
      <ImageBackground 
        resizeMode="cover"
        style={styles.image}
        source={imageProps}
      />
    </View>
  )
};

export default Card;
