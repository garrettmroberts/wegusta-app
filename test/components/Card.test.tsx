import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';

import Card from '../../components/Card/Card';

describe('<DecisionButton />', () => {
  it('runs a local image correctly', () => {
    const { toJSON } = render(
      <Card imageProps={require('../../assets/sample-image.jpeg')} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders a remote image correctly', () => {
    const { toJSON } = render(
      <Card imageProps={{ uri: 'https://picsum.photos/200/300' }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
