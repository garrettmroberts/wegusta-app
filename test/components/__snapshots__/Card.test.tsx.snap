// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DecisionButton /> renders a remote image correctly 1`] = `
<View
  style={
    Object {
      "height": 1034,
      "padding": 10,
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "borderRadius": 20,
        "flex": 1,
        "overflow": "hidden",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "uri": "https://picsum.photos/200/300",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;

exports[`<DecisionButton /> runs a local image correctly 1`] = `
<View
  style={
    Object {
      "height": 1034,
      "padding": 10,
      "position": "absolute",
      "width": 750,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "borderRadius": 20,
        "flex": 1,
        "overflow": "hidden",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
